nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -632,22 +627,35 @@ in
       }
     ];
 
+    systemd.tmpfiles.rules = [
+      "d '${cfg.stateDir}' 0750 ${cfg.user} ${cfg.group} - -"
```

Comment:
This could be handled by `StateDirectory` from `systemd.exec`.

Reply:
Unfortunately not:
> The specified directory names must be relative, and may not include "..".

Reply:
Through `LogsDirectory` cannot be specified the full path `/var/spool/nginx`

Reply:
Why does this have to be in `/var/spool`? Is it just for backwards compatibility?

Reply:
/var/spool/nginx directory is not only used to keep logs, but also hashed data and temporary files. On highly loaded systems amount of hashed data may be big. Therefore, placing of them in /var/log/nginx directory is incorrect.

Reply:
Why does the directory where you save them make any difference?

Let's keep the discussion about logs in the other comment/thread btw.